examples/pylint/W0127_self_assigning_variable.py

```diff
@@ -0,0 +1,5 @@
+i = 0
+while i < 10:
```

Comment:
You don't need a while loop here. Even something like `x = 1`, followed by `x = x` should cause this error.